src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java

```diff
@@ -765,7 +765,17 @@ protected final JsonNode _fromFloat(JsonParser p, DeserializationContext ctxt,
         if (nt == JsonParser.NumberTypeFP.BIG_DECIMAL) {
             return _fromBigDecimal(ctxt, nodeFactory, p.getDecimalValue());
         }
-        if (ctxt.isEnabled(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)) {
+
+        // [databind#4801] Add JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS
+        DatatypeFeatures dtf = ctxt.getDatatypeFeatures();
+        boolean useBigDecimal;
+        if (dtf.isExplicitlySet(JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS)) {
+            useBigDecimal = dtf.isEnabled(JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS);
+        } else {
+            useBigDecimal = ctxt.getConfig().isEnabled(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS);
```

Comment:
I think there's short-cut as `ctxt.isEnabled(DeserializationFeature)`, will change if there is.